### ISSUE TITLE
Markdown support for accordion details content

### ISSUE THREAD

**Issue by PriyavKaneria** [cotes2020/jekyll-theme-chirpy#2046]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### Is your feature request related to a problem? Please describe

When using the accordion markdown using `<details>` and `<summary>` tags the markdown inside is not parsed and rendered by jekyll.

### Describe the solution you'd like

A plugin or support for parsing the contents of a details/summary using custom syntax possibly for proper rendering

### Describe alternatives you've considered

Current workaround is using html inside the content of accordion, but that defeats the purpose of using markdown

**Comment by PriyavKaneria** [comment]
I have made a plugin for resolving this issue, raising PR

**Comment by kungfux** [comment]
That's correct. However, as far as I can remember, you cannot hide the code highlight inside the spoiler because the markdown is not being processed. So it's use is kind of limited.

**Comment by cotes2020** [comment]
Hi @kungfux,

> you cannot hide the code highlight inside the spoiler because the markdown is not being processed.

Can you give us a simple example of this?

**Comment by PriyavKaneria** [comment]
added some reproduction example in the issue description if that helps

**Comment by cotes2020** [comment]
Surround your markdown code with `<div markdown="1">` and `</div>`, problem solved, no plugin needed.

**Comment by kungfux** [comment]
> Surround your markdown code with `<div markdown="1">` and `</div>`, problem solved, no plugin needed.

What kind of magic spell is this? 🙃

**Comment by PriyavKaneria** [comment]
> Surround your markdown code with `<div markdown="1">` and `</div>`, problem solved, no plugin needed.

do I need some additional config for this? never knew about this but if it works it's best. anyway this is not working for me right now as is

<img width="418" alt="image" src="https://github.com/user-attachments/assets/deb82ce0-66ce-4123-beb7-5e2ff9281379">

<img width="826" alt="image" src="https://github.com/user-attachments/assets/12ef5c48-b5cb-48f7-985a-d7566ffd5209">

**Comment by PriyavKaneria** [comment]
when I use it inside the details tag like so
<img width="367" alt="image" src="https://github.com/user-attachments/assets/64798acd-366c-45a5-9c06-bd2e4df02760">

it stomehow works for the same line but nothing after the line break
<img width="823" alt="image" src="https://github.com/user-attachments/assets/4182fd66-d0e7-4ae9-ba56-9b94a018b923">

am I still doing something silly?

also, true I did not think about markdown portability. but the image syntax for setting width height and shadows are similar as well `{: w="700" h="400" .shadow}` so maybe it might be fine. your call anyway

**Comment by cotes2020** [comment]
Indentation is treated as a code block in Markdown, so you need to remove all starting whitespace.

<img src="https://github.com/user-attachments/assets/2891914b-1478-4617-b8a1-e1593a559921" width="500">

**Comment by PriyavKaneria** [comment]
yes this works. cool, turns out there is a lot about markdown I'm not aware of! Thanks
I'll close this issue and PR
For anyone who still wants to use a plugin, check out the [PR](https://github.com/cotes2020/jekyll-theme-chirpy/pull/2048/files#diff-c026a9fcff0430bf75eba4aa43e40bc5942d826a5ba5b465232e55e3c284393a)
and use it as in this [PR description](https://github.com/cotes2020/jekyll-theme-chirpy/pull/2048#issue-2668714307)

**Comment by huanyushi** [comment]
> @PriyavKaneria, write it this way:
> 
> ```
> <details>
>   <summary>summary</summary>
>   <div markdown="1">
>   
> ### Heading
>   
>   1. Foo
>   2. Bar
>      * Baz
>      * Qux
> 
> ### Some Javascript
>   
> ```js
> function logSomething(something) {
>   console.log('Something', something);
> }
> ``` 
>   </div>
> </details>
> ```

A better choice:

```html

<details markdown="1">
<summary>summary</summary>
  
### Heading
  
  1. Foo
  2. Bar
     * Baz
     * Qux

### Some Javascript
  
```js
function logSomething(something) {
  console.log('Something', something);
}
``` <!-- intentionally added fourth quote so I can nest it under a code block -->

</details>

```

You can just add `markdown="1"` as an attribute rather than a tag.
